File: Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_SpriteData.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_SpriteData
*/
public class t_SpriteData : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public uint	m_id;
	public string	m_sShader;
	public t_SpriteDataOne[]	m_aData;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_SpriteData() {
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_id = 0;
		m_sShader = "";
		m_aData = new t_SpriteDataOne[0];
	}
	public bool read(CReadVariable cVariable) {
		try {
			m_id = cVariable.getU32();
			cVariable.getString(ref m_sShader,255);
			{
				int n = (int) cVariable.getU32();
				if (n > 65536) {
					cVariable.error("array size error in m_aData");
					return false;
				}
				m_aData = new t_SpriteDataOne [n];
				for(int i = 0;i < n;++i) {
					m_aData[i] = new t_SpriteDataOne();
					if (!m_aData[i].read(cVariable)) return false;
				}
			}
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(m_id);
		cVariable.put(ref m_sShader,255);
		{	
				if (m_aData.Length > 65536) {
					cVariable.error("array size error in m_aData");
					return false;
				}
			int n = m_aData.Length;
			cVariable.put((uint) n);
			for(int i = 0;i < n;++i) {
				if (!m_aData[i].write(cVariable)) return false;

			}
		}
		return true;
	}
};

File: Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_SpriteData.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_SpriteDataOne import	t_SpriteDataOne
#/==========================================================================
#*!
#	@brief	t_SpriteData
#/
class t_SpriteData:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_id = 0	#U32
	m_sShader = ""	#std::string
	m_aData = []	#t_SpriteDataOne
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_id = 0
		self.m_sShader = ""
		self.m_aData = []

	def read(self,cVariable):
		self.m_id = cVariable.getU32()
		self.m_sShader = cVariable.getString(255)
		n = cVariable.getU32()
		if (n > 65536) :
			cVariable.error('array size error in m_aData')
			return False
		self.m_aData = [None]*n
		for i in range(n):
			self.m_aData[i] = t_SpriteDataOne()
			if (not self.m_aData[i].read(cVariable)):
				return False
		return True

	def write(self,cVariable):
		cVariable.putU32(self.m_id)
		cVariable.putString(self.m_sShader,255)
		if (len(self.m_aData) > 65536) :
			cVariable.error('array size error in m_aData')
			return False
		n = len(self.m_aData)
		cVariable.putU32(n)
		for i in range(n):
			if (not self.m_aData[i].write(cVariable)):
				return False
		return True

